### PR TITLE
Fix demultiplex

### PR DESCRIPTION
**Pull request by JureZmrzlikar**

### CODE REVIEW

File: iCount/files/fastq.py [hunk 1]
```diff
@@ -1,213 +1,59 @@
...
-
-
-def _sl2l(s):
-    """TODO."""
-    # return ''.join((_Dsl2l[ij] for ij in s))
-    return s
-
-
-def _determine_format(fname):
-    """
```
Comment (tomazc) [inline iCount/files/fastq.py]: @JureZmrzlikar why did you remove this function? It is very important, because it assures that after demultiplexing, all quality scores will be in the right range (Phred scale) so that cutadapt will work correctly...

Comment (JureZmrzlikar) [inline iCount/files/fastq.py]: Because it was not being used.

Comment (tomazc) [inline iCount/files/fastq.py]: Of course it was used by the `reader` function, see [here](https://github.com/tomazc/iCount/pull/141/files#diff-2cddde0c74a339b441c85efa31dc23acL97). I'm not sure why check is set to `False`, should be `True` by default.
I suggest you move the function `_determine_format` into the fastq module, and call it whenever format needs to be known. For example, by the `cutadapt` function.

Comment (tomazc) [inline iCount/files/fastq.py]: Can you please split this PR into three separate:
- "Restructure fastq module" and "Fix if xlsite is on begging of chrom" into "fastq and xlsites: minor fixes"
- "Restructure fastq module" into "fastq: restructure", should include the `_determine_format` functionallity.
- "Refactor demultiplex" into "demultiplex: refactoring"

(please try to follow the formatting of PR titles, i.e., module: description)

Comment (JureZmrzlikar) [inline iCount/files/fastq.py]: Thans for comments @tomazc , I think i've addressed them. Please, have another review pass.

**Comment by JureZmrzlikar** [comment]
Closes #140

File: iCount/files/fastq.py (hunk 1 above)
Comment (tomazc) [inline iCount/files/fastq.py]: @JureZmrzlikar I suggested you implement [option 1](https://github.com/tomazc/iCount/pull/141#discussion_r123047603). Why did you decide for 2?

Comment (JureZmrzlikar) [inline iCount/files/fastq.py]: While it is possible that users come up with their own, already demultiplexed data, we cannot be sure that their data was demultiplexed in "iCount compatible" way (appending random barcode into the headers). Besides that, i think it is better to transform quality encodings once for all and not worry about it in any later step.

**Comment by tomazc** [comment]
Please, fix title of this PR.

**Comment by JureZmrzlikar** [comment]
By adding "closes XY"?

File: iCount/files/fastq.py (hunk 1 above)
Comment (tomazc) [inline iCount/files/fastq.py]: Such checks (for proper barcode encoding) should be added where needed.
Please, implement option 1 as it is more general and more user friendly.

**Comment by tomazc** [comment]
By following the [Contributing guidelines](http://icount.readthedocs.io/en/latest/contributing.html#commit-messages).
